File: UNXNavigatorDemo/Base/Controllers/BaseNavigationController.m
//
//  BaseNavigationController.m
//  UNXNavigatorDemo
//
//  Created by Leo Lee on 2020/10/29.
//

#import "BaseNavigationController.h"

@interface BaseNavigationController ()

@end

@implementation BaseNavigationController

- (UIViewController *)childViewControllerForStatusBarStyle {
    return self.topViewController;
}

- (UIViewController *)childViewControllerForStatusBarHidden {
    return self.topViewController;
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Common/Comtrollers/BaseNavigationController.swift
//
//  BaseNavigationController.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class BaseNavigationController: UINavigationController {
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override var childForStatusBarStyle: UIViewController? {
        return topViewController
    }
    
    override var childForStatusBarHidden: UIViewController? {
        return topViewController
    }

}
